### DISCUSSION TITLE
Add Documentation for the aead Module

### DISCUSSION THREAD

**Discussion by MythicByte** [briansmith/ring#2177] [General]
Add examples for the aead module. Provide more information about the API. The section on the nonce and the trait that needs to be implemented should be better explained, and it should include an example.

**Comment by briansmith** [comment]
I definitely agree we need to document the API. (Un)fortunately, I'm hoping to "real soon now" add a new AEAD API that will be easier for people to understand. Thus, I've not spent effort documenting the current API.
